internal/handlers/pg/msg_find.go

```diff
@@ -50,11 +50,13 @@ func (h *Handler) MsgFind(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, er
 		return nil, err
 	}
 
-	if params.MaxTimeMS != 0 {
-		ctxWithTimeout, cancel := context.WithTimeout(ctx, time.Duration(params.MaxTimeMS)*time.Millisecond)
-		defer cancel()
+	username, _ := conninfo.Get(ctx).Auth()
 
-		ctx = ctxWithTimeout
+	cancel := func() {}
```

Comment:
I'm thinking that it's easy to forget calling `cancel()` on newly added error, and leave resources behind. Maybe this is not a concern as we will cleanup unused cursors.

Alternative could be always cancelling cursor upon any error something like this, but it's not ideal as `cancel()` shouldn't be cancelled if cursor is created.
```
defer func(err error){
if err !=nil{
cancel()
}
}(err)
```

Reply:
`go vet` catches those problems:

```
-> task lint
task: [lint] bin/golangci-lint run --config=.golangci.yml
internal/handlers/pg/msg_find.go:59:3: lostcancel: the cancel function is not used on all paths (possible context leak) (govet)
                ctx, cancel = context.WithTimeout(ctx, time.Duration(params.MaxTimeMS)*time.Millisecond)
                ^
internal/handlers/pg/msg_find.go:72:4: lostcancel: this return statement may be reached without using the cancel var defined on line 59 (govet)
                        return nil, err
                        ^
task: Failed to run task "lint": exit status 1
```

Reply:
Good to know, yay for linting :+1: